File: WebApplication/ClientApp/src/app/_orchestration/athleteResultBase.ts
import { PaceWithTime } from "../_core/enums/paceWithTime";

export interface AthleteResultBase {
  athleteId: number
  fullName: string
  raceAge: number
  genderAbbreviated: string
  bib: string
  paceWithTimeCumulative: PaceWithTime
}

File: Api/Orchestration/AthleteResultBase.cs
using Api.DataModels;

namespace Api.Orchestration;

public abstract class AthleteResultBase(
    int athleteId,
    string fullName,
    DateTime dateOfBirth,
    Gender gender,
    string bib,
    DateTime courseStartTime,
    PaceWithTime paceWithTimeCumulative)
{
    public int AthleteId { get; } = athleteId;
    public string FullName { get; } = fullName;
    public int RaceAge { get; } = DateTimeHelper.GetRaceAge(dateOfBirth, courseStartTime);
    public string GenderAbbreviated { get; } = gender.ToAbbreviation();
    public string Bib { get; } = bib;
    public PaceWithTime PaceWithTimeCumulative { get; } = paceWithTimeCumulative;
}
